src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -118,8 +118,17 @@ public static function createConnection($dsn, array $options = array())
         $params += $options + self::$defaultConnectionOptions;
         if (null === $params['class'] && !\extension_loaded('redis') && !class_exists(\Predis\Client::class)) {
             throw new CacheException(sprintf('Cannot find the "redis" extension, and "predis/predis" is not installed: %s', $dsn));
+        } elseif (isset($params['predis_options']) && !class_exists(\Predis\Client::class)) {
+            throw new CacheException(sprintf('DSN contains "predis_options" and "predis/predis" is not installed: %s', $dsn));
```

Comment:
The option can come from the $options argument, so the message should not tell about the DSN I'd say.
What about `Invalid connection option: "predis_options" provided but "predis/predis" is not installed.', $dsn));